kubernetes/utils/deserialize.py

```diff
@@ -0,0 +1,285 @@
+    # Replace CamelCased action_type into snake_case
+    kind = data["kind"]
+    kind = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", kind)
+    kind = re.sub("([a-z0-9])([A-Z])", r"\1_\2", kind).lower()
+    # Expect the user to load namespaced objects more often
+    if hasattr(k8s_api, "read_namespaced_{0}_with_http_info".format(kind)):
+        fnc_lookup = "read_namespaced_{0}_with_http_info".format(kind)
+    elif hasattr(k8s_api, "read_{0}_with_http_info".format(kind)):
+        fnc_lookup = "read_{0}_with_http_info".format(kind)
+    # Try with the create if no read fnc found
+    elif hasattr(k8s_api, "create_{0}_with_http_info".format(kind)):
+        fnc_lookup = "create_{0}_with_http_info".format(kind)
+    else:
+        fnc_lookup = None
+
+    info = {
+        "group": group,
+        "version": version,
+        "kind": kind,
+        "fcn_to_call": fcn_to_call,
+        "fnc_lookup": fnc_lookup,
+    }
+    if verbose:
+        print("Lookup info: {}".format(info))
+
+    if fnc_lookup:
+        if hasattr(k8s_api, fnc_lookup):
+            # Get response_type from func pydoc
+            response_type = _find_response_type(getattr(k8s_api, fnc_lookup))
+            if verbose:
+                print(
+                    "Lookup function found: {} in k8s_api: {} response_type: "
+                    "{} lookup info: {}".format(
+                        fnc_lookup,
+                        k8s_api,
+                        response_type,
+                        info))
+        else:
+            if verbose:
+                print(
+                    "Lookup function not found: {} in k8s_api: {} "
+                    "response_type: {} lookup info: {}".format(
+                        fnc_lookup,
+                        k8s_api,
+                        response_type,
+                        info))
+    else:
+        msg = "Failed to find a function to inspect; lookup info: {}".format(
+            info)
+        raise FailToLoadError(reason=msg)
+
+    return response_type
+
+
+def _find_response_type(func):
+    for line in pydoc.getdoc(func).splitlines():
+        if line.startswith(PYDOC_RETURN_LABEL):
+            return line[len(PYDOC_RETURN_LABEL):].strip()
+    return ""
+
+
+def load_from_dict(data, klass=None, verbose=False):
+    """
+    Load object/objects from a dictionary containing valid kubernetes
+    API object (i.e. List, Service, etc).
+
+    Input:
+    data: a dictionary holding valid kubernetes objects
+    klass: class literal for deserialized object,
+    or string of class name.
+    verbose: If True, print additional info.
+        Default is False.
+    Raises:
+        FailToLoadError which holds list of load failures.
+    """
+    # If it is a list type, will need to iterate its items
+    load_exceptions = []
+    obj = None
+    if "List" in data["kind"]:
+        obj_list = []
+        # Could be "List" or "Pod/Service/...List"
+        # This is a list type. iterate within its items
```

Comment:
I think api_client.deserialize can handle list types, e.g. `V1DeploymentList`. Why do we need to break the list down?

Reply:
I think this an excellent observation; I will explore this.

Reply:
I think returning ObjectsList will work however there is one edge case where we need to account for multiple types of objects.
The following json  can have multiple objects with different kinds.

```kubectl get all --all-namespaces -o json > all.objs.json```

I guess in this case we need to traverse the list items and extract the different kinds of objects then we can output a list of ObjectsList. I think that should be the only time we return a python list with multiple K8s ObjectsList.

Reply:
`kubectl get all` sends multiple requests to the apiserver (you can see them with `kubectl get all -v=6`). This python client doesn't have a `get all` utility. If it does, for each request the response will be a single ObjectsList.

Reply:
I think in this situation I was thinking more of the json file output from kubectl  "all.objs.json" more than the way kubectl interacts with the API; in this context I was trying to support loading such files (with multiple object kinds) via the load_from_json() function.

Please see this https://github.com/kubernetes-client/python/issues/977#issuecomment-543009440 for some additional info.

Reply:
It would be nice if we can return ObjectsList when the input is indeed a single ObjectsList. ObjectsList's are standard API classes, and a user should be able to give a "ObjectsList" as the klass hint. 

For the edge case with `kubectl get all`, I see it as a lower priority because:
- `kubectl get all` is a broken command. The term `all` is misleading and what people think it does is usually wrong: https://github.com/kubernetes/kubectl/issues/151. Currently it is suggested to not use `kubectl get all` and use `kubectl api-resources` instead.
- I haven't seen lots of requests for deserializing a list of objects with different kinds. And I think it can be broken down by iterating through the list.

I'm okay with adding a utility for deserializing `all.objs.json`, as long as we support the standard ObjectList kinds e.g. `V1DeploymentList`

Reply:
Sound good @roycaihw I agree with the priorities you have outlined. Thank you for your insight.